src/java.base/share/classes/java/lang/invoke/VarHandles.java

```diff
@@ -106,66 +107,66 @@ else if (type == double.class) {
             }
         }
         else {
-            // TODO This is not lazy on first invocation
-            // and might cause some circular initialization issues
-
-            // Replace with something similar to direct method handles
-            // where a barrier is used then elided after use
-
-            if (UNSAFE.shouldBeInitialized(refc))
-                UNSAFE.ensureClassInitialized(refc);
+            if (UNSAFE.shouldBeInitialized(refc)) {
```

Comment:
This seems unrelated to the issue this PR is really about (if I understand correctly). Would it make sense to address this as part of another PR?

Reply:
Also, perhaps @PaulSandoz knows more history on this one (I believe this code predates FFM).